lib/App/index.js

```diff
@@ -608,53 +605,37 @@ class App {
   }
 
   async preprocess() {
-    Log(colors.green('✓ Pre-processing app...'));
-
-    const appJson = await this._getAppJsonFromFolder();
-    if (!appJson) {
-      throw new Error('This app.json is invalid!');
-    }
+    App.getManifest({ appPath: this.path });
 
-    this._isValidAppJson(appJson);
+    Log(colors.green('✓ Pre-processing app...'));
 
     if (App.hasHomeyCompose({ appPath: this.path })) {
       await HomeyCompose.build({ appPath: this.path });
     }
   }
 
   async version(version) {
-    let appJsonPath;
-    let appJson;
+    let appJsonPath = path.join(appPath, 'app.json');
 
-    if( App.hasHomeyCompose({ appPath: this.path }) ) {
-        let appJsonComposePath = path.join(this.path, '.homeycompose', 'app.json');
-        let exists = false;
-        try {
-          await accessAsync(appJsonComposePath, fs.constants.R_OK | fs.constants.W_OK);
-          exists = true;
-        } catch( err ) {}
+    const composeAppJsonPath = path.join(appPath, '.homeycompose', 'app.json');
+    if (fs.existsSync(composeAppJsonPath)) {
```

Comment:
Why not use `App.hasHomeyCompose()` here? Perhaps icw `getManifest()`

Reply:
Mostly because it actually needs to know the path of the manifest to write it back out again...